src/main/java/com/pss/ProjectileSimulationSuite.java

```diff
@@ -5,59 +5,86 @@
 import com.pss.handlers.*;
 
 import javax.vecmath.Vector3d;
+import java.util.ArrayList;
+import java.util.List;
 
+/**
+ * This class provides a suite for running projectile simulations.
+ * The simulations steps are computed, stored, and output to various formats.
+ */
 public class ProjectileSimulationSuite {
-    private static Vector3d[] _results;
+
+    private static int MAX_SIMSTEPS = 100;
+    private static int steps;
     private static IOutputResults[] _resultsOutputers;
-    private static final int MAX_OUTPUTERS = 4;
-    // private static final int MAX_OUTPUTERS = 3;
-    private static final int MAX_SIMSTEPS = 80;
-    private static IGetConfiguration _configHandler;
     private static ProjectileSimulator _simulator;
+    private static List<Vector3d> _results;
+    private static Vector3d[] _sim_steps;
 
+    /**
+     * The main method of the suite. It initializes the simulation, runs it,
+     * and then outputs the results.
+     *
+     * @param args Command line arguments
+     */
     public static void main(String[] args) {
         initSimulation();
 
         for (int t = 0; t < MAX_SIMSTEPS; t++) {
-            _results[t] = new Vector3d(0, 0, 0);
-            _results[t].add(_simulator.updatePosition());
+            _sim_steps[t] = new Vector3d(0, 0, 0);
+            _sim_steps[t].add(_simulator.updatePosition());
+            // Increment steps counter if the z coordinate is non-negative
+            if (_sim_steps[t].z >= 0)
+                steps++;
+        }
```

Comment:
Could we add an `else { t = MAX_SIMSTEPS; }` after the ground is "hit" so this loop stops executing if the results aren't used later? Or do we need _sim_steps to fill out, even if the values aren't used?

Reply:
Done